src/main/resources/templates/image-resizer.html

```diff
@@ -1,84 +1,148 @@
+            <select id="presetSelect" name="presetSelect" class="form-control">
+              <option value="">Select resolution</option>
+              <option value="640,480">640 x 480</option>
+              <option value="800,600">800 x 600</option>
+              <option value="1024,768">1024 x 768</option>
+              <option value="1280,720">1280 x 720</option>
+              <option value="1920,1080">1920 x 1080</option>
+              <option value="2560,1440">2560 x 1440</option>
+              <option value="3840,2160">3840 x 2160</option>
+            </select>
+          </div>
+
+          <button type="submit" class="btn btn-primary" id="resizeButton">
+            <span class="spinner-border d-none" id="loadingSpinner" role="status"></span>
+            <span id="buttonText">Resize Image</span>
+          </button>
+        </div>
+      </form>
     </div>
-    <div class="mt-2">
-      <a th:href="${resizedImage}" download="resized-image" class="btn btn-success">Download Image</a>
+
+    <!-- Right column - Image preview -->
+    <div class="resize-preview">
+      <div th:if="${resizedImage}" class="image-result">
+        <div class="image-controls">
+          <div class="mouse-button-viewer">
+          </div>
+        </div>
+
+        <div class="image-container" id="imageContainer">
+          <div class="cursor-position" id="cursorPosition"></div>
+          <img th:src="${resizedImage}" alt="Resized image" class="img-fluid" id="resizedImageOutput">
+        </div>
+        <div class="download-button-wrapper">
+          <button class="btn btn-sm" id="downloadBtn" title="Download Image">
+             <span>Download</span>
+          </button>
+          <span id="resolution" th:text="'Resolution: ' + ${originalWidth} + ' x ' + ${originalHeight}"></span>
+        </div>
+      </div>
+
+      <div th:unless="${resizedImage}" class="placeholder-message">
+        <p>Your resized image will appear here</p>
+      </div>
     </div>
   </div>
-  
-  <script>
-    document.getElementById('resizeForm').onsubmit = function() {
-      const fileInput = document.getElementById('imageFile');
-      const file = fileInput.files[0];
-      
-      if (!file) {
-        alert('Please select an image file');
-        return false;
-      }
-      
-      if (file.size > 10 * 1024 * 1024) {
-        alert('File size exceeds the maximum limit of 10MB');
-        return false;
-      }
-      
-      // Show loading indicator
-      document.getElementById('loadingIndicator').style.display = 'block';
-      document.getElementById('resizeButton').disabled = true;
-      
-      return true;
-    };
-    
-    // Reset loading state if we're back on the page after form submission
-    if (performance.navigation.type === 1) { // 1 is PAGE_RELOAD
-      document.getElementById('loadingIndicator').style.display = 'none';
-      document.getElementById('resizeButton').disabled = false;
-    }
+
+
+  <script th:inline="javascript">
```

Comment:
You should probably use the page specific js system we have to load the script from a separate file. It also automatically defers the script to after page load, so the `DOMContentLoaded` event is unnecessary.